app/controllers/items/search_code.js

```diff
@@ -1,44 +1,112 @@
-import Ember from 'ember';
-import SearchCode from '../search_code';
-import AjaxPromise from 'stock/utils/ajax-promise';
-const { getOwner } = Ember;
+import Ember from "ember";
+import SearchCode from "../search_code";
+import capitalize from "lodash/capitalize";
 
 export default SearchCode.extend({
   item: null,
   inlineDescription: true,
   store: Ember.inject.service(),
+  messageBox: Ember.inject.service(),
+  packageService: Ember.inject.service(),
+  i18n: Ember.inject.service(),
 
-  allPackageTypes: Ember.computed("fetchMoreResult", 'item.isSet', function(){
-    if(this.get('item.isSet')) {
-      return this.get('item.setItem.code').allChildPackagesList();
+  allPackageTypes: Ember.computed("fetchMoreResult", "item.isSet", function() {
+    if (this.get("item.isSet")) {
+      return this.get("item.setItem.code").allChildPackagesList();
     } else {
-      return this.store.query('code', { stock: true });
+      return this.store.query("code", { stock: true });
     }
   }),
 
+  deleteAndAssignNew(packageType) {
+    const item = this.get("item");
+    const detailType = item.get("detailType");
+    const detailId = item.get("detail.id");
+    this.showLoadingSpinner();
+    this.get("packageService")
+      .deletePackage(detailType, detailId)
+      .then(() => {
+        if (this.isSubformPackage(packageType)) {
+          this.assignNew(packageType);
+        } else {
+          this.assignNew(packageType, { deleteDetailId: true });
+        }
+      });
+  },
+
+  warnAndAssignNew(pkgType) {
+    const existingPkgType = this.get("item.code");
+    const packageName = existingPkgType.get("name");
+    const newPackageName = pkgType.get("name");
+    const translation = this.get("i18n");
+
+    this.get("messageBox").custom(
+      translation.t("items.new.subform.delete_subform_waring", {
+        newPackageName,
+        packageName
+      }),
+      translation.t("not_now"),
+      null,
+      translation.t("continue"),
+      () => {
+        this.deleteAndAssignNew(pkgType);
+      }
+    );
+  },
+
+  isSubformPackage(packageType) {
+    return !!packageType.get("subform");
```

Comment:
@abulsayyad123 , it's better to check `["computer", "computer_accessory", "electrical"].indexOf(packageType.get("subform"))` because some of the packageTypes have subform field as "medical".